File: 23_oct/229. Majority Element II/229. Majority Element II.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        freq = dict()
        for n in nums: 
            if freq.get(n, -1)==-1:
                freq[n]=0
            freq[n]+=1
        
        ans = []
        for k, v in freq.items(): 
            if v>len(nums)//3: 
                ans.append(k)
        return ans

File: 23_oct/229. Majority Element II/229. Majority Element II.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        unordered_map<int, int> freq;
        vector<int> ans;
        for(auto n : nums) freq[n]++;
        int n = nums.size();
        for (auto [k, v] : freq){
            if(v>n/3) ans.push_back(k);
        }
        return ans;
    }
};
